pkg/controller/initializer.go

```diff
@@ -0,0 +1,76 @@
+package controller
+
+import (
+	"github.com/kiegroup/kie-cloud-operator/pkg/apis/app/v1"
+	"github.com/kiegroup/kie-cloud-operator/pkg/controller/kieapp"
+	"github.com/kiegroup/kie-cloud-operator/pkg/controller/kieapp/logs"
+	oappsv1 "github.com/openshift/api/apps/v1"
+	buildv1 "github.com/openshift/api/build/v1"
+	oimagev1 "github.com/openshift/api/image/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logs.GetLogger("kie-cloud-operator.controller")
+
+func init() {
+	// AddToManagerFuncs is a list of functions to create controllers and add them to a manager.
+	addManager := func(mgr manager.Manager) error {
+		k8sService := GetInstance(mgr)
+		reconciler := kieapp.KieAppReconciler{Service: &k8sService}
+		return add(mgr, &reconciler)
+	}
+	AddToManagerFuncs = []func(manager.Manager) error{addManager}
+}
+
+func add(mgr manager.Manager, reconciler reconcile.Reconciler) error {
```

Comment:
does moving this up one level make it more difficult to add future controllers? I feel like the way its currently organized we have more flexibility around what each controller watches/loads. i know we currently only have 1 (kieapp) but maybe in the future we offer an upgrade controller or something?

Reply:
hmm, is it likely that we'll add another controller? We could move the initializer.go file one level down, but the init() will have to stay in this package to happen in time, and we'll end up with a Go file with one/two lines in it again.

I was trying to put myself in the shoes of someone reading the code for the first/second time. It's really hard to understand what happens and in what order, even for me sometimes. Maybe it's ok to have an initializer both up here and down there, even if the top one only has a few lines. WDYT?

Reply:
i think that might better, so we can confine the controller-specific watches/logic to each controller package the operator might offer in the future. thoughts?